### PR TITLE
[WIP] fix isEmpty for truthy numbers

### PR DESCRIPTION
**Pull request by faliah**
Hello there, I noticed that `isEmpty` returns `true` for all numbers, but test description for truthy numbers says `returns false with truthy number`. On the other hand docs provide the following function signature:
```
isEmpty :: a -> Boolean: Empty Object, Array or String
```
which might be a good reason to throw error or return `false` for any type other than `Object`, `Array` or `String`

I also have a question regarding falsey numbers, namely `0`, is there any reasoning why it should return `false` there?

### CODE REVIEW

**Comment by dalefrancis88** [comment]
I think we should also consider NaN here as well. I think it should be considered empty but what are your thoughts?

**Comment by evilsoft** [comment]
Thanks so much for the submission!

So this is a tricky one, now that we throw `Monoid`s in the mix. I am more a fan of reasonable defaults as opposed to throwing errors. So I was thinking `true` would be reasonable as single element values are always empty, as they cannot contain anything. But maybe that was the wrong thinking.

So for instance `0` is only an empty when it is paired with a `Sum` Monoid. If true does not seem like a reasonable default, maybe we throw on anything that is not `null`, `undefined`, `NaN`, `Monoid`, `String` or `Array`.

Thoughts?

**Comment by evilsoft** [comment]
`NaN`, `undefined` and `null` are all isomorphic to `Unit` and are the values for the Trivial Monoid.
So it would make sense that those return true.

**Comment by faliah** [comment]
@dalefrancis88 and @evilsoft thanks for such a quick review and thorough explanations!
I thought about it again and decided to also check the how the alternatives from Ramda and lodash work with these single element values and it turns out that crocks is actually doing the same thing as lodash:

```
|isEmpty(_)|Crocks|Lodash|Ramda|
|---|---|---|---|---|
|NaN|true|true|false|
|undefined|true|true|false|
|null|true|true|false|
|1|true|true|false|
|0|true|true|false|
|-1|true|true|false|
|true|true|true|false|
|false|true|true|false
```
I ran tests and generated this table here: https://repl.it/@faliah/Empty-checks

Ramda's implementation follows this reasoning:
```
Returns true if the given value is its type's empty value; false otherwise.
```
But I find it confusing that `null` and `NaN` are not considered empty values of their types in this case.

So now after a bit of thinking over I do agree to these 
> So I was thinking true would be reasonable as single element values are always empty, as they cannot contain anything. But maybe that was the wrong thinking.
and these
> NaN, undefined and null are all isomorphic to Unit and are the values for the Trivial Monoid.
So it would make sense that those return true.
thoughs and I would revert back to true for all numbers, and update the method description in the doc. How does it sound?

**Comment by dalefrancis88** [comment]
I think that would be a good move, I was just coming back to add similar thoughts to these after not quite feeling comfortable with it. Really good research on the other libraries, nice follow ups :)

File: src/core/isEmpty.js [hunk 1]
```diff
@@ -16,6 +16,10 @@ function isEmpty(x) {
     return !Object.keys(x).length
   }
 
+  if (typeof x === 'number') {
```
[the remark this answers is not in the document]
Comment (faliah) [reply src/core/isEmpty.js]: Decided to revert the logic back

**Comment by faliah** [comment]
@dalefrancis88 thanks, the pointers were very useful and enlightening indeed. So here is the updated PR

**Comment by evilsoft** [comment]
@faliah if you would like do add your self to the contributors list for all your hard work and research on this, you should be able to run this on your local branch:
`yarn contributors:add faliah doc`
This will add you to the readme.

**Comment by faliah** [comment]
@evilsoft oh cool, it works, added myself to contributors!
Huge thanks for such an opportunity, it's my very first OSS contribution, so I am pretty excited :octocat:

**Comment by dalefrancis88** [comment]
Nice work @faliah there are lots of other pieces that need to be worked on. If you'd like to do more, just let us know :)

**Comment by faliah** [comment]
@dalefrancis88 yeah sure, I'll look through the issues once I'll have some time, thanks for all your help again!

**Pull request merged**